astropath/bayesian.py

```diff
@@ -0,0 +1,190 @@
+    """
+    Raw prior for a given set of Pchance values and/or n(m)
+
+    Args:
+        Pchance (float or np.ndarray):
+            Chance probability
+        Sigma_m (float or np.ndarray):
+            Number density of sources on the sky brighter than m
+        method (str):
+            inverse :: Assign inverse to Sigma_m
+            inverse1 :: Assign inverse to Sigma_m * half_light
+            inverse2 :: Assign inverse to Sigma_m * half_light**2
+            orig_inverse :: Assign inverse to P_chance
+            identical :: All the same
+            linear :: 1-Pchance (not recommended)
+        half_light (float or np.ndarray, optional):
+            Angular size of the galaxy
+            Only required for several methods
+
+    Returns:
+        float or np.ndarray:
+
+    """
+    if method == 'linear':
+        return 1 - Pchance
+    elif method == 'orig_inverse':
+        return 1. / Pchance
+    elif method == 'inverse':
+        return 1. / Sigma_m
+    elif method == 'inverse1':
+        return 1. / Sigma_m / half_light
+    elif method == 'inverse2':
+        return 1. / Sigma_m / half_light**2
+    elif method == 'identical':
+        return np.ones_like(Pchance)
+    else:
+        raise IOError("Bad method {} for prior_Oi".format(method))
+
+
+def pw_Oi(r_w, theta_half, theta_prior, scale_half=1.):
+    """
+    Calculate p(w|O_i) for a given galaxy
+
+    Args:
+        r_w (np.ndarray):
+            offset from galaxy center in arcsec
+        theta_half (float):
+            Half-light radius of this galaxy in arcsec
+        theta_prior (dict):
+            Parameters for theta prior
+        scale_half (float):
+
+    Returns:
+        np.ndarray: Probability values; un-normalized
+
+    """
+    p = np.zeros_like(r_w)
+    ok_w = r_w < theta_prior['max']*theta_half
+    if theta_prior['method'] == 'core':
+        # Wolfram
+        norm = theta_half * np.log(theta_prior['max']+1)
+        if np.any(ok_w):
+            p[ok_w] = theta_half / (r_w[ok_w] + theta_half) / norm
+    elif theta_prior['method'] == 'uniform':
+        norm = theta_half * theta_prior['max']
+        if np.any(ok_w):
+            p[ok_w] = 1. / norm
+    elif theta_prior['method'] == 'exp':
+        norm = theta_half*scale_half * (scale_half - (
+                scale_half+theta_prior['max'])*np.exp(-theta_prior['max']/scale_half))
+        if np.any(ok_w):
+            p[ok_w] = (r_w[ok_w] / theta_half) * np.exp(-r_w[ok_w]/(scale_half*theta_half)) / norm
+    else:
+        raise IOError("Bad theta method")
+    #
+    if norm == 0:
+        raise ValueError("You forgot to normalize!")
+    # Return
+    return p
+
+
+def px_Oi(box_radius, frb_coord, eellipse, cand_coords,
+          theta_prior, step_size=0.1, return_grids=False):
+    """
+    Calculate p(x|O_i), the primary piece of the analysis
+
+    Main concept:
+        1. Set an area to analyze
+        2. Discretize it to 0.1"
+        3. Convolve
```

Comment:
I think it would be useful to elaborate on this a bit more.

Reply:
tried